Public/Src/Engine/Processes/SharedOpaqueOutputLogger.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using BuildXL.Pips.Operations;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Collections;
+using JetBrains.Annotations;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Responsible for keeping a log of all paths written to by a given process pip.
+    /// 
+    /// Paths are flushed to an underlying file as soon as they're reported via <see cref="RecordFileWrite(PathTable, AbsolutePath)"/>.
+    /// 
+    /// A number of root directories may be set in which case the log will record only 
+    /// those paths that fall under one of those directories.  The typical use case is to
+    /// use all shared opaque directory outputs of a process as root directories.
+    /// </summary>
+    /// <remarks>

```

Comment:
><remarks> [](start = 8, length = 9)

Can you make some remarks here stating that no  path table can be used here because the class has to be made serializable for VM execution.

Reply:
will do